client-id/handler.go

```diff
@@ -0,0 +1,107 @@
+package clientid
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/SKF/go-rest-utility/problems"
+	"github.com/SKF/go-utility/v2/stages"
+
+	middleware "github.com/SKF/go-enlight-middleware"
+	"github.com/SKF/go-enlight-middleware/client-id/enforcement"
+	"github.com/SKF/go-enlight-middleware/client-id/extractor"
+	"github.com/SKF/go-enlight-middleware/client-id/models"
+	custom_problems "github.com/SKF/go-enlight-middleware/client-id/problems"
+	"github.com/SKF/go-enlight-middleware/client-id/store"
+)
+
+type Middleware struct {
+	Tracer middleware.Tracer
+
+	Stage models.Environment
+
+	Extractor   extractor.Extractor
+	Enforcement enforcement.Policy
+	Store       store.Store
+}
+
+type (
+	ClientID = models.ClientID
+	Store    = store.Store
+)
+
+func New(opts ...Option) *Middleware {
+	m := &Middleware{
+		Tracer: new(middleware.OpenCensusTracer),
+
+		Stage: stages.StageProd,
```

Comment:
I'm hesitant if we should allow instantiating Middleware without specifying stage.

Reply:
With the building pattern used in this and the other new middleware, this was the best approach I could find.

Like you said it will be set by most consumers, but if you forget we default to the production environment which should be the most restricted of all.